standard/namespace.lua

```diff
@@ -7,11 +7,21 @@
 --
 
 local Class = require('Module:Class')
+local FnUtil = require('Module:FnUtil')
+local Table = require('Module:Table')
 
 local Namespace = {}
 
 function Namespace.isMain()
 	return mw.title.getCurrentTitle():inNamespace(0)
 end
 
+Namespace.getIdsByName = FnUtil.memoize(function()
+	return Table.map(mw.site.namespaces, function(id, ns) return ns.name, id end)
+end)
+
+function Namespace.idFromName(name)
+	return Namespace.getIdsByName()[name]
```

Comment:
`return mw.title.new(name ..':dummy').namespace`
Is an alternative solution, if you want to do some performance testing on which method is faster.

Reply:
I did some quick testing - `mw.site.namespaces` is faster than `mw.title.new(name ..':dummy').namespace`

`mw.site.namespaces` takes ~20us to generate the mapping (for all namespaces).

`mw.title.new(name ..':dummy').namespace` takes 100us for each namespace

Reply:
Solid 👍